### PR TITLE
feat:(bookmark-component) added bookmark side sheet component and updated cli 

### PR DESCRIPTION
**Pull request by Noggling**
In this pull request: 
-  New bookmark side sheet is added.
-  At tis stage this is not coded as a widget but should be converted in the future.  
-  Updated the cli header with header form EDS
-  Updated to bookmark related modules as needed

### CODE REVIEW

File: packages/react/components/bookmark/src/utils/append-bookmark-to-uri.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export function appendBookmarkIdToUrl(id: string): string {
```
Comment (odinr) [inline packages/react/components/bookmark/src/utils/append-bookmark-to-uri.ts:1]: change to const

**Review by odinr — APPROVED** [review]

File: packages/modules/bookmark/src/client/bookmarkClient.ts [hunk 2]
```diff
@@ -228,11 +233,17 @@ export class BookmarkClient {
     }
 
     public async addBookmarkFavoriteAsync(bookmarkId: string): Promise<void> {
-        this.#bookmarkAPiClient.addFavorite('v1', { bookmarkId });
+        const response = await this.#bookmarkAPiClient.addFavorite('v1', { bookmarkId });
```
[the remark this answers is not in the document]
Comment (odinr) [reply packages/modules/bookmark/src/client/bookmarkClient.ts:236]: see https://github.com/equinor/fusion-framework/issues/796

File: packages/react/components/bookmark/src/components/import-bookmark/ImportBookmark.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    return (
+        <Dialog style={{ width: '600px' }} open={isOpen}>
+            <Dialog.Header>Import bookmark</Dialog.Header>
+            <Dialog.Content>
+                <p>This bookmark was created by {currentBookmark?.createdBy.name}</p>
+                <p>Would you like to import it?</p>
+            </Dialog.Content>
+            <Dialog.Actions>
+                <div style={{ display: 'flex', gap: '0.2em' }}>
```
[the remark this answers is not in the document]
Comment (Noggling) [reply packages/react/components/bookmark/src/components/import-bookmark/ImportBookmark.tsx:26]: i know, just a slipup when copying code :P hehe

File: packages/modules/bookmark/src/client/bookmarkClient.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Noggling) [reply packages/modules/bookmark/src/client/bookmarkClient.ts:236]: I think this is suited for its own pull request

File: packages/react/components/bookmark/src/hooks/useBookmarkGrouping.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            .filter((v, i, a) => a.indexOf(v) === i)
+            .map((groupingProperty) => ({
+                groupingProperty,
+                values: array
+                    .filter((s) => getKey(s) === groupingProperty)
+                    .filter((s) => {
+                        const fieldData = s[field];
+                        if (typeof fieldData === 'string') {
+                            return searchText ? fieldData.includes(searchText) : true;
+                        }
```
[the remark this answers is not in the document]
Comment (Noggling) [reply packages/react/components/bookmark/src/hooks/useBookmarkGrouping.ts:48]: I agree, will fix

**Pull request merged**